server/bundler/extensions-loader.js

```diff
@@ -0,0 +1,34 @@
+const existsSync = require( 'fs' ).existsSync;
+const path = require( 'path' );
+const camelCase = require( 'lodash' ).camelCase;
+
+function getReducers( extensions = {} ) {
+	return extensions
+		.filter( ( extension ) => existsSync( path.join( 'client', 'extensions', extension, 'state', 'reducer.js' ) ) )
+		.map( ( extension ) => `'${ camelCase( extension ) }': require( 'extensions/${ extension }/state/reducer' )\n` );
+}
+
+function getExtensionsModule( extensions ) {
+	return `module.exports = {
+		reducers: function() {
+			return {
+				${ getReducers( extensions ) }
+			};
+		}
+	};
+`;
+}
+
+module.exports = function( content ) {
+	this.cacheable && this.cacheable();
```

Comment:
It's a webpack cacheable thing. It's identical to the one in `server/bundler/loader.js`
Docs here: https://webpack.github.io/docs/how-to-write-a-loader.html#flag-itself-cacheable-if-possible